principal/server.go

```diff
@@ -238,29 +254,30 @@ func (s *Server) loadTLSConfig() (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
-// Listener returns the listener of Server s
-func (s *Server) Listener() *Listener {
+// ListenerForE2EOnly returns the listener of Server s
```

Comment:
For all the `*E2EOnly` field changes here:

When I initially saw these struct fields, I noticed these fields are not exported from the struct, which implies they are for internal package use only. 

BUT, then I noticed we are making the contents of these fields available by exporting these functions. For example, `Queues()` could be called to retrieve the internal queues reference of the principal.

Finally, I noticed that they were only being used for the E2E tests, so I have renamed these functions to allow them to still be used by E2E tests, but to make it obvious they are not for general use.

(If you were expecting these functions WOULD be available for general use, I'm happy to revert this part of the PR)

Reply:
Good points there, great observations. I think at some point in time we made use of at least the `Listener` function, but since then move forward do something else.

I guess I'm fine with renaming those functions for now.